Sources/MapboxNavigation/InstructionsBannerView.swift

```diff
@@ -4,65 +4,85 @@ import MapboxCoreNavigation
-     */
-    func didTapInstructionsBanner(_ sender: BaseInstructionsBannerView) {
-        logUnimplemented(protocolType: InstructionsBannerViewDelegate.self, level: .debug)
-    }
+ A banner view that contains the current step instruction and responds to tap and swipe gestures.
+ 
+ This class responds and gets updated as the user progresses along a route according to the `NavigationComponent` and `BaseInstructionsBannerView` protocol.
+*/
+@IBDesignable
+open class InstructionsBannerView: BaseInstructionsBannerView, NavigationComponent {
     
     /**
-     `UnimplementedLogging` prints a warning to standard output the first time this method is called.
+     Updates the instructions banner info as the user progresses along a route.
+     
+     - parameter service: The `NavigationService` instance that passes the instruction.
+     - parameter instruction: The `VisualInstructionBanner` instance to be presented.
+     - parameter routeProgress: The`RouteProgress` instance that the instruction banner view is updating.
      */
-    func didSwipeInstructionsBanner(_ sender: BaseInstructionsBannerView, swipeDirection direction: UISwipeGestureRecognizer.Direction) {
-        logUnimplemented(protocolType: InstructionsBannerViewDelegate.self, level: .debug)
-    }
-}
-
-/// :nodoc:
-@IBDesignable
-open class InstructionsBannerView: BaseInstructionsBannerView, NavigationComponent {
     public func navigationService(_ service: NavigationService, didPassVisualInstructionPoint instruction: VisualInstructionBanner, routeProgress: RouteProgress) {
         update(for: instruction)
     }
 }
 
-/// :nodoc:
+/**
+ A banner view that contains the current step instruction along a route and responds to tap and swipe gestures, as the base of `InstructionsCardView` and `InstructionsBannerView`.
+*/
 open class BaseInstructionsBannerView: UIControl {
+    
+    /**
+     A view that contains a simple image indicating a type of maneuver.
+     */
     public weak var maneuverView: ManeuverView!
+    
+    /**
+     A primary instruction label indicates the current step.
+     */
     public weak var primaryLabel: PrimaryLabel!
+    
+    /**
+     A secondary instruction label below the `primaryLabel` to provide detail information of current step.
+     */
     public weak var secondaryLabel: SecondaryLabel!
+    
+    /**
+     A styled lable indicates the remaining distance along the current step.
+     */
     public weak var distanceLabel: DistanceLabel!
+    
+    /**
+     A vertical view divides between `maneuverView` and `distanceLabel` to `primaryLabel` and `secondaryLabel`.
+     */
     public weak var dividerView: UIView!
     weak var _separatorView: UIView!
+    
+    /**
+     A invisible helper view for visualizing the result of the constraints.
+     */
     public weak var separatorView: SeparatorView!
+    
+    /**
+     A drag view indicates whether there're more steps in the remaining route.
```

Comment:
Not sure I follow naming here. What is `drag` view?

Reply:
Because we could have a drag gesture of that to show more step instructions? As described in the following:
https://github.com/mapbox/mapbox-navigation-ios/blob/68b2c5711d3a52cc792010de90a1a087a5644826/Sources/MapboxNavigation/InstructionsBannerView.swift#L318-L319

Reply:
Ah, I see. Let's maybe mention that: `A view, which indicates that there're more steps in the current route. This view can be swiped to the bottom to see all of these remaining steps.`

Reply:
Thanks for the suggestion, I've updated this part to provide the use case.